src/include/storage/sql_table.h

```diff
@@ -231,6 +231,17 @@ class SqlTable {
    */
   ProjectionMap ProjectionMapForOids(const std::vector<catalog::col_oid_t> &col_oids);
 
+  /**
+   * Clears the contents of this table and reinitializes it
+   */
+  void Reset();
```

Comment:
Who is it used by, what purpose does it serve, and when is it not safe to use it? This seems like a blunt instrument ripe for abuse in the future that'll cause races between the garbage collector and the table.